### DISCUSSION TITLE
插件仓库网页的一个实现

### DISCUSSION THREAD

**Discussion by AnzhiZhang** [MCDReforged/PluginCatalogue#313] [Ideas]
## 背景

- 当前我个人制作了一个网页版的插件仓库，实现了插件的搜索和筛选功能，及更美观和符合直觉的信息展示。网址为：https://mcdr.zhanganzhi.com
- [仓库链接](https://github.com/AnzhiZhang/MCDReforgedWebsite)
- 该网站部署在 Vercel，使用我个人的账号；投票/点赞 功能数据库位于国内服务器
- #150

## 建议

现征求 MCDR 核心开发者及社区关于以下想法的意愿和建议：

1. 该项目的任何改进建议
2. 将该项目仓库移到 MCDReforged 组织中
3. 在插件库文档加入该网页，并作为用于查找插件首选建议
4. 修改项目的版权声明，当前在网站和文档声明并非官方或社区项目，为个人项目
5. 项目预留了 MCDR 官方网站功能的空间，可用于展示文档、信息、教程或提供相应链接
6. Vercel 部署的相关问题
7. 域名费用我可以个人继续承担，无需担心

**Comment by alex3236** [comment]
1. 使用完整的 MCDR Logo 很不美观，因为几乎无法辨认——尝试在不包含文字的图标后添加矢量字体，我认为此处不必使用 Minecraft 风格。
![image](https://user-images.githubusercontent.com/45303195/229333899-057e6ea2-92e7-4d00-9fce-10562e767387.png)
2. 左侧功能区没有明显地划分目的性。看起来现有逻辑是根据 Modrinth 制定的，但此处情况不同——插件页面没有包括完整自述文件，因此用户必须转到仓库页面了解更多信息，仓库链接非常重要。根据现有情况，我认为：上方的容器用于存放插件信息，下方存放安装和下载相关功能。作者和仓库链接应当被移动至上方。
3. “安装”一栏应当提供帮助消息，说明该指令来源。例如，可以设置鼠标悬停显示的帮助窗。
4. Server side rendering 是真的慢（
![image](https://user-images.githubusercontent.com/45303195/229334281-62df4a1a-0a7b-4a35-8262-1559a360364c.png)

**Comment by Fallen-Breath** [comment]
对于第三方的插件仓库转正，最重要的因素有这么几点：

- 可靠性。该网站是否稳定。这包括网站的托管服务提供商、域名、数据库等方面，它们需要能够长期稳定地运行和维护
- 可用性。该网站对用户的可用性如何。要考虑到用户的网络环境（如国内的GFW问题）、使用语言（中英文）等因素
- 可维护性。该网站应具备长期（以年计）的维护能力，包括但不限于在插件仓库发生变更、上游依赖需要重大升级、网站托管服务出现变更等情况下，能够及时进行维护的能力
- 准确性。该网站的数据来源应当稳定、可靠且不易被篡改，展示的内容应具备实时性和准确性

若无法做到，该网站只能作为一个可供参考的第三方网站

至于具体的实现，这方面得等上述要点明确后，再继续确定

**Comment by AnzhiZhang** [comment]
> 对于第三方的插件仓库转正，最重要的因素有这么几点：
> 
> * 可靠性。该网站是否稳定。这包括网站的托管服务提供商、域名、数据库等方面，它们需要能够长期稳定地运行和维护
> * 可用性。该网站对用户的可用性如何。要考虑到用户的网络环境（如国内的GFW问题）、使用语言（中英文）等因素
> * 可维护性。该网站应具备长期（以年计）的维护能力，包括但不限于在插件仓库发生变更、上游依赖需要重大升级、网站托管服务出现变更等情况下，能够及时进行维护的能力
> * 准确性。该网站的数据来源应当稳定、可靠且不易被篡改，展示的内容应具备实时性和准确性
> 
> 若无法做到，该网站只能作为一个可供参考的第三方网站
> 
> 至于具体的实现，这方面得等上述要点明确后，再继续确定

- 可靠性：Vercel 是较大商业公司部署，且没有任何重大变更的可能性，无需担心可靠性。域名当前有效期为一年，目前我可以承诺 3 年的保证，域名注册商为 Cloudflare。
- 可用性：
  - 如 @alex3236 所述，会有一定延迟，但国内的可访问性是没有问题的，延迟问题有优化空间和可能性，且在部分情况下可超过 GitHub 的响应速度。
  - 国际化完善，自动识别用户语言，且有更多语言的国际化空间。
  - 针对无光标输入设备的残障人士做了可访问性设计，针对视觉残障人士作了部分无障碍设计，具有基于机器朗读的访问能力。
- 可维护性：我个人可以承担维护责任，如无重大意外（重大个人事业或家庭变故、自然灾害、战争等）情况，目前可以提供3-5年的保证。如有更多社区用户可以加入维护工作，将可以进一步提升维护性的稳定性
- 准确性：直接使用该插件仓库的数据。关于数据同步情况，有二层缓存，更新时间为10分钟

**Comment by AnzhiZhang** [comment]
@alex3236 所述问题已全部解决，关于安装说明，由于未来会有内置插件管理器，所以我认为暂无需修改。

**Comment by alex3236** [comment]
https://github.com/MCDReforged/PluginCatalogue/discussions/312#discussioncomment-8305197
